### ISSUE TITLE
Accuracy of storageObject and storage types.

### ISSUE THREAD

**Issue by cmhhelgeson** [three-types/three-ts-types#929]
Within StorageBufferNode.d.ts, the storage and storageObject functions are typed like so...
```
export const storage: (
    value: ArrayLike<number>,
    nodeOrType: NodeOrType,
    count: number,
) => ShaderNodeObject<StorageBufferNode>;
export const storageObject: (
    value: ArrayLike<number>,
    nodeOrType: NodeOrType,
    count: number,
) => ShaderNodeObject<StorageBufferNode>;
```

However, in the latest WebGPU compute examples, these functions seem to mostly be provided with StorageBufferAttributes as their first arguments. For instance, in the WebGPU Compute Points example, storage nodes are initialized with StorageBufferAttributes:

```
const particleBuffer = new StorageInstancedBufferAttribute( particleNum, particleSize );
const velocityBuffer = new StorageInstancedBufferAttribute( particleNum, particleSize );

const particleBufferNode = storage( particleBuffer, 'vec2', particleNum );
const velocityBufferNode = storage( velocityBuffer, 'vec2', particleNum );
````

Although the last two arguments correspond to a NodeOrType and a number type respectively, StorageInstancedBufferAttribute does not correspond to the ArrayLike type. Accordingly, when attempting to replicate the usage of storage or storageObject in the most recent WebGPU compute examples, Typescript indicates that the StorageBufferAttributes are not assignable to this parameter type.

<img width="304" alt="image" src="https://github.com/three-types/three-ts-types/assets/62450112/05d162a7-371d-4874-bce6-7c6f888b3766">

**Comment by cmhhelgeson** [comment]
Looking through the code for storage, storageObject, and storageBufferNode, I think it might just be a general typing error? The code for StorageBufferNode and all the class it inherits from don't seem to act as if value is necessarily an ArrayLike<number>...

**Comment by Methuselah96** [comment]
Yeah, the types for the Nodes are still pretty rough around the edges (including some incorrect types). I'll probably take a look at fixing this sometime this month or feel free to create a PR yourself.

**Comment by Methuselah96** [comment]
I went ahead and fixed this in #931. I'm not super sure about the types in this area of the code, but it does fix this particular problem. It should go out with r165 at the end of the month.
